### PR TITLE
API: Adds DeletePrefixes

### PR DESCRIPTION
**Pull request by JoshVanL**
Adds a new DeletePrefixes API which deletes all jobs (as well as counters) whose names match the given prefixes. Note the given strings to this API are prefixes of the _job names_, not the underlying ETCD keys. Prefixes will be serialized into the actual ETCD keys.

This is useful for Dapr to add a feature whereby jobs (of all of types) in a particular namespace (some job prefix) are deleted when a Kubernetes Namespace is deleted.

Moves `/internal/tests` to `/tests` to expose cron ETCD helper funcs for library consumers. Expands package with a `cron` sub-package to introduce single and clustered cron server setup helpers.

Exposes the `CounterGarbageCollectionInterval` time duration pointer option to enable testing time interval triggered counter garbage collection. This is set to a low value during some tests, which has a side effect of soft testing this doesn't prematurely delete counters.

Slight refactor of `api_test.go` to reside in a different package to prevent a circular import from `/tests`.

Adds `/fake` for exposing a fake cron that library consumers can use for testing.

### CODE REVIEW

File: api.go [hunk 1]
```diff
@@ -103,8 +102,6 @@ func (c *cron) Delete(ctx context.Context, name string) error {
 	if _, err := c.client.Delete(ctx, jobKey); err != nil {
 		return err
 	}
-	counterKey := c.key.CounterKey(name)
```
Comment (cicoyle) [inline api.go:106]: Why is this being removed? Won't this cause counters to linger even when a job is removed?

File: api.go [hunk 2]
```diff
@@ -114,17 +111,54 @@ func (c *cron) Delete(ctx context.Context, name string) error {
...
+
+		c.queueCache.Delete(jobKey)
+		errs = append(errs, c.queue.Dequeue(jobKey))
+	}
+
+	for _, prefix := range prefixes {
+		keyPrefix := c.key.JobKey(prefix)
+		c.log.V(3).Info("deleting jobs with prefix", "prefix", keyPrefix)
+
+		resp, err := c.client.Delete(ctx, keyPrefix, clientv3.WithPrefix(), clientv3.WithPrevKV())
```
Comment (cicoyle) [inline api.go:152]: Don't we need to cleanup the counters here as well or am I missing that somewhere in the diff?

File: cron.go [hunk 3]
```diff
@@ -318,6 +339,7 @@ func (c *cron) handleInformerEvent(ctx context.Context, e *informer.Event) error
 
 	defer c.queueLock.DeleteUnlock(string(e.Key))
 	c.queueCache.Delete(string(e.Key))
+	c.collector.Push(c.key.CounterKey(c.key.JobName(e.Key)))
```
Comment (cicoyle) [inline cron.go:342]: oh I see it here now

**Review by cicoyle — APPROVED** [review]

File: cron.go (hunk 3 above)
Comment (JoshVanL) [reply cron.go:342]: :+1:

**Review by artursouza — APPROVED** [review]

**Review by artursouza — CHANGES_REQUESTED** [review]
Please, target main branch.

**Comment by artursouza** [comment]
This change is too big for a patch. We can release 0.3

**Comment by JoshVanL** [comment]
@artursouza makes sense- I have opened [this PR](https://github.com/diagridio/go-etcd-cron/pull/38) to fast forward master to keep up with release-0.2

**Review by artursouza — APPROVED** [review]

**Review by artursouza — APPROVED** [review]

**Pull request merged**